plugins/azure-sites-backend/src/service/router.ts

```diff
@@ -52,28 +66,116 @@ export async function createRouter(
     '/:subscription/:resourceGroup/:name/start',
     async (request, response) => {
       const { subscription, resourceGroup, name } = request.params;
-      console.log('starting...');
-      response.json(
-        await azureSitesApi.start({
-          subscription,
-          resourceGroup,
-          name,
-        }),
+      const token = getBearerTokenFromAuthorizationHeader(
+        request.header('authorization'),
       );
+      const entityRef = request.body.entityRef;
+      if (typeof entityRef !== 'string') {
+        throw new InputError('Invalid entityRef, not a string');
+      }
+      const entity = await catalogApi.getEntityByRef(entityRef, { token });
+
+      if (entity) {
+        const annotationName =
+          entity.metadata?.annotations?.[AZURE_WEB_SITE_NAME_ANNOTATION];
+        if (
+          annotationName &&
+          !(await azureSitesApi.validateSite(annotationName, name))
+        ) {
+          throw new NotFoundError();
+        }
+
+        const decision = permissions
+          ? (
+              await permissions.authorize(
+                [
+                  {
+                    permission: azureSitesActionPermission,
+                    resourceRef: entityRef,
+                  },
+                ],
+                {
+                  token,
+                },
+              )
+            )[0]
+          : undefined;
+        if (decision && decision.result === AuthorizeResult.DENY) {
+          throw new NotAllowedError('Unauthorized');
+        }
+
+        logger.info(
+          `entity ${entity.metadata.name} - azure site "${name}" is starting...`,
+        );
+        response.json(
+          await azureSitesApi.start({
+            subscription,
+            resourceGroup,
+            name,
+          }),
+        );
+      } else {
+        throw new NotFoundError();
+      }
     },
   );
   router.post(
     '/:subscription/:resourceGroup/:name/stop',
     async (request, response) => {
       const { subscription, resourceGroup, name } = request.params;
-      console.log('stopping...');
-      response.json(
-        await azureSitesApi.stop({
-          subscription,
-          resourceGroup,
-          name,
-        }),
+      const token = getBearerTokenFromAuthorizationHeader(
+        request.header('authorization'),
       );
+
+      const entityRef = request.body.entityRef;
+      if (typeof entityRef !== 'string') {
+        throw new InputError('Invalid entityRef, not a string');
+      }
+      const entity = await catalogApi.getEntityByRef(entityRef, { token });
+
+      if (entity) {
+        const annotationName =
+          entity.metadata.annotations?.[AZURE_WEB_SITE_NAME_ANNOTATION];
+
+        if (
+          annotationName &&
+          !(await azureSitesApi.validateSite(annotationName, name))
+        ) {
+          throw new NotFoundError();
```

Comment:
Wonder if we should be throwing some additional data here for when `validateSite` fails compared to when we don't get the `entity` from the Catalog because it doesn't exists?